pkg/mount/magic.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2014-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mount
+
+import "fmt"
+
+// This code does not use a map. Why?
+// The name and magic pairs are stored as simple tuples, rather
+// than a map, because these kinds of maps are rarely searched
+// more than once, and the size of such a map in the busybox
+// is much larger than a simple array of structs.
+// These values are from Linux 5.4. FS magics change very
+// slowly over time, so there is no generate script.
+// This is not tagged for any one kernel as these values
+// are (or should be) universal.
+var magics = []struct {
+	magic uint32
+	name  string
+}{
+	{magic: 0x00c36400, name: "ceph"},
+	{magic: 0xa501fcf5, name: "vxfs"},
+	{magic: 0x65735543, name: "fuse_ctl"},
+	{magic: 0x65735546, name: "fuse"},
+	{magic: 0x482b, name: "hfsplus"},
+	{magic: 0x20030528, name: "orangefs"},
+	{magic: 0x24051905, name: "ubifs"},
+	{magic: 0xadf5, name: "adfs"},
+	{magic: 0xadff, name: "affs"},
+	{magic: 0x5346414f, name: "afs"},
+	{magic: 0x0187, name: "autofs"},
+	{magic: 0x73757245, name: "coda"},
+	{magic: 0xf15f, name: "ecryptfs"},
+	{magic: 0x414a53, name: "efs"},
+	{magic: 0xe0f5e1e2, name: "erofs"},
+	// This is what Linux has: three filesystems with the same magic.
+	// HMMM.
```

Comment:
Perhaps, in this case, the `MagicToName` function can return a `[]string` in order of priority, `ext4`, then `ext3`, then `ext2`.

Reply:
I thought of that but it's a bunch of nonsense for one corner case. I don't think it's worth it. In general, systems should have ext4 and that will cover the bases.